doc/website/advanced/best-practice-for-testing.md

```diff
@@ -0,0 +1,257 @@
+- overflow with `operator+`, e.g. adding `7` and `8`
+- underflow with `operator-`, e.g. subtracting `8` from `7`
+
+Here, ZOMBIES comes into play and gives us some guide to identify this cases.
+
+Some of this test might made it necessary to define the behavior of the class in the first place, e.g. defining that an `operator+` will saturate to the max value instead of overflowing.
+
+## How To Do It Better
+
+At first, the test is split into multiple test cases according the AAA pattern
+
+```c++
+TEST_F(SingleDigitNumber_test, DefaultConstructedObjectIsCorrectlyInitialized)
+{
+    constexpr uint64_t EXPECTED_VALUE{0U};
+
+    SingleDigitNumber sut;
+
+    EXPECT_EQ(static_cast<uint64_t>(sut), EXPECTED_VALUE);
+}
+```
+
+The test also has a meaningful name. If this fails in the CI, it is immediately clear what is broken.
+Additionally, the tested object is called `sut`. This makes it easy to identify the actual test object.
+Lastly, a `constexpr` is used for the expected value. This removes a magic value and also makes the output of a failed test more readable, since it is immediately clear what's the actual tested value and what's the expected value.
+
+Now let's continue with further tests, applying the ZOMBIES principle
+
+```c++
+TEST_F(SingleDigitNumber_test, ConstructionWithValidValueCreatesNumberWithSameValue)
+{
+    constexpr uint64_t NUMBER_VALUE{7U};
+    constexpr uint64_t EXPECTED_VALUE{NUMBER_VALUE};
+
+    SingleDigitNumber sut{NUMBER_VALUE};
+
+    EXPECT_EQ(static_cast<uint64_t>(sut), EXPECTED_VALUE);
+}
+
+TEST_F(SingleDigitNumber_test, ConstructionWithInvalidValueResultsInSaturation)
+{
+    constexpr uint64_t NUMBER_VALUE{42U};
+    constexpr uint64_t EXPECTED_VALUE{9U};
+
+    SingleDigitNumber sut{NUMBER_VALUE};
+
+    EXPECT_EQ(static_cast<uint64_t>(sut), EXPECTED_VALUE);
+}
+
+TEST_F(SingleDigitNumber_test, AddingZeroDoesNotChangeTheNumber)
+{
+    constexpr SingleDigitNumber NUMBER{3U};
+    constexpr SingleDigitNumber ZERO{0U};
+    constexpr uint64_t EXPECTED_VALUE{3U};
+
+    auto sut = NUMBER + ZERO;
+
+    EXPECT_EQ(static_cast<uint64_t>(sut), EXPECTED_VALUE);
+}
+
+TEST_F(SingleDigitNumber_test, AddingOneIncreasesTheNumberByOne)
+{
+    constexpr SingleDigitNumber NUMBER{3U};
+    constexpr SingleDigitNumber ONE{1U};
+    constexpr uint64_t EXPECTED_VALUE{4U};
+
+    auto sut = NUMBER + ONE;
+
+    EXPECT_EQ(static_cast<uint64_t>(sut), EXPECTED_VALUE);
+}
+
+// and so on
+```
+
+These are some examples showing how to apply the ZOMBIES principle to find good test cases.
+They also exert all the good practices mentioned previously, like clear and distinct names or avoiding magic numbers.
+
+# Slightly More Advanced Topics
+
+## Typed Tests
+
+TODO:
+- simple example which can be used as template
+
+## Parameterized Tests
+
+TODO:
+- simple example which can be used as template
+- https://www.sandordargo.com/blog/2019/04/24/parameterized-testing-with-gtest
+- use `std::tie`
+
+## Mocks
+
+TODO:
+- just a few links
```

Comment:
Everything in https://github.com/google/googletest/tree/release-1.10.0/googlemock/docs